tests/test_script_helper.py

```diff
@@ -225,318 +287,312 @@ def generate_random_synapses_locations(cell,syn_filename,num_syns=10000,seed=2):
+
     for sec in cell.apical:
         tot_L += sec.L
-    print("basal_L,tot_L",basal_L,tot_L)
-    for i in range(num_syns):
+
+    for _ in range(num_syns):
         L_counter = 0
-        x_L = random_obj.uniform(0,tot_L)
-        if x_L<basal_L:
-            for ix,sec in enumerate(list(cell.basal)):
-                if L_counter+sec.L>x_L:
-                    x = (x_L - L_counter)/sec.L
-                    syn_L.append(["basal",ix,x])
+        x_L = random_obj.uniform(0, tot_L)
+        if x_L < basal_L:
+            for ix, sec in enumerate(list(cell.basal)):
+                if L_counter + sec.L > x_L:
+                    x = (x_L - L_counter) / sec.L
+                    syn_L.append(["basal", ix, x])
                     break
                 else:
-                    L_counter+=sec.L
+                    L_counter += sec.L
         else:
-            x_L -=basal_L
-            for ix,sec in enumerate(list(cell.apical)):
-                if L_counter+sec.L>x_L:
-                    x = (x_L - L_counter)/sec.L
-                    syn_L.append(["apical",ix,x])
+            x_L -= basal_L
+            for ix, sec in enumerate(list(cell.apical)):
+                if L_counter + sec.L > x_L:
+                    x = (x_L - L_counter) / sec.L
+                    syn_L.append(["apical", ix, x])
                     break
                 else:
-                    L_counter+=sec.L
+                    L_counter += sec.L
 
     print(syn_filename)
-    with open(syn_filename,'w') as f:
+    with open(syn_filename, 'w') as f:
         for syn in syn_L:
-            f.write("[%s,%s,%s]"%(syn[0],syn[1],syn[2])+"\n")
-
-def set_up_parameters(orig_morphology_file, model_file, frequency, manual_total_nsegs, synapse_file):
-    ''' Sets the program's arguments as global parameters for the entire testing process '''
-
+            f.write("[%s,%s,%s]\n" % (syn[0], syn[1], syn[2]))
 
 
 def loadtemplate(template_file):
-    ''' Check if the template is already defined, if it does not defined it load it, if it is defined it does not reload the template'''
-    
+    '''load template if not already defined'''
     model_obj_name = template_file.split(".")[0].split('/')[-1]
-    if h.name_declared(model_obj_name)==0:
-        print("loading template '" + model_obj_name +"'")
+    if h.name_declared(model_obj_name) == 0:
+        print("loading template '%s'" % model_obj_name)
         h.load_file(template_file)
     else:
-        print("WARNING The template '" +  model_obj_name + "' is already defined... not loading.")
+        print("WARNING The template '%s' is already defined... not loading." % model_obj_name)
 
 
-def change_cell_location(control_cell, soma_list):
-    if soma_list:
-        control_cell.soma[0].push() 
-    else:
-        control_cell.soma.push()
-    
-    for i in range(0,int(h.n3d())):
-        h.pt3dchange(i, 150+h.x3d(i), 150+h.y3d(i), 150+h.z3d(i), h.diam3d(i))
+def change_cell_location(soma, offset=150):
```

Comment:
is this needed? I tried running the tests without changing the location, and they seemed to pass.  I would like to remove it if I could

Reply:
It is not needed for the simulation, but it helps to debug.  (It will make sure that the two cells are not in the exact location)

Reply:
ah, ok, I will leave it, but add a comment